bin/source-erlang.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing,
+#   software distributed under the License is distributed on an
+#   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#   KIND, either express or implied.  See the License for the
+#   specific language governing permissions and limitations
+#   under the License.
+
+# Build Erlang from source on systems where desired package
+# versions are not available
+#
+# While these scripts are primarily written to support building CI
+# Docker images, they can be used on any workstation to install a
+# suitable build environment.
+
+# stop on error
+set -e
+
+# Check if running as root
+if [[ ${EUID} -ne 0 ]]; then
+  echo "Sorry, this script must be run as root."
+  echo "Try: sudo $0 $*"
+  exit 1
+fi
+
+. ${SCRIPTPATH}/detect-os.sh
+
+redhats='(rhel|centos|fedora)'
+debians='(debian|ubuntu)'
+
+# Install per-distro dependencies according to:
+#  http://docs.basho.com/riak/1.3.0/tutorials/installation/Installing-Erlang/
+# NB: Dropping suggested superfluous packages; fop and unixodbc-dev
+if [[ ${ID} =~ ${redhats} ]]; then
+    yum install git gcc glibc-devel make ncurses-devel openssl-devel autoconf
+elif [[ ${ID} =~ ${debians} ]]; then
+    apt-get update
+    apt-get install -y git build-essential autoconf libncurses5-dev openssl libssl-dev xsltproc
```

Comment:
So when installing libssl version 1.0.2 I receive the following error:

```
dpkg: regarding libssl1.0-dev_1.0.2n-1ubuntu5.1_arm64.deb containing libssl1.0-dev:arm64:
 libcurl4-openssl-dev:arm64 conflicts with libssl1.0-dev
  libssl1.0-dev:arm64 (version 1.0.2n-1ubuntu5.1) is to be installed.

dpkg: error processing archive libssl1.0-dev_1.0.2n-1ubuntu5.1_arm64.deb (--install):
 conflicting packages - not installing libssl1.0-dev:arm64
```

It's conflicting with `libcurl4-openssl-dev` which is installed in `bin/apt-dependencies.sh`

How would you like me to move forward?

Reply:
Hm. We only use `libcurl` for building our test harness for our JavaScript tests, which are slowly being converted over to Elixir. Once that is done, we won't need that library anymore.

Our choices are:

1. Build without `libcurl` support in `couchjs` and forego the `make javascript` test suite. This seems like a bad idea since those tests regularly find problems when porting the software. However, it would be a short-term problem, since the Elixir test suite will take over soon enough.
2. Build without SSL support in Erlang, and thus CouchDB. You would only be able to speak to CouchDB on port 5984, not the optional 6984 port. This is probably less desirable than 1.
3. Complain if the combination of OS and Erlang fails and exit with a non-zero status code. Maybe also print something helpful about needing a newer version of Erlang that will link against libssl1.1. Users of `build.sh` will have to know to specify Erlang 20+ when we are compiling Erlang from source.

My opinion at the moment - keeping in mind I have just released CouchDB 2.3.0-RC1 - is #3. Let's take the easy way out.